### PR TITLE
added consul wrapper

### PR DESCRIPTION
**Pull request by DayFan**

### CODE REVIEW

File: consul/consul_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package consul
+
+import (
+	"testing"
+
+	"github.com/hashicorp/consul/testutil"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFoo_bar(t *testing.T) {
```
Comment (cv21) [inline consul/consul_test.go]: :)

**Review by cv21 — APPROVED** [review]

File: consul/consul.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+		Name:    srvName,
+		Address: localIP.String(),
+		Port:    svcPort,
+		Check: &api.AgentServiceCheck{
+			CheckID: srvName,
+			TTL:     checkTTL,
+		},
+	}
+
+	if err = client.Register(srvRegInfo); err != nil && client.IsReachable() {
```
Comment (avdva) [inline consul/consul.go]: if the client is not Reachable, and we failed to register, we return `client, nil`. is the Client in valid state in that case?
this condition looks very suspicious

Comment (DayFan) [reply consul/consul.go]: This behavior is mentioned in a description of the function. If registration failed we can still use `Client` for register service again. If the client can't reach the local agent, we still can use a function to update TTL(yes, this function return error) and this behavior similar to the original Consul package.

Comment (avdva) [reply consul/consul.go]: Then, what is the point of registration in `NewDefaultClient`?
I'm an important service, I  want to be registered. The return value won't tell me anything about it.
So, I must to call Register myself, or check, if I'm registered.
If registration isn't important for me,all the more so, why calling it there?

**Review by avdva — APPROVED** [review]

**Pull request merged**